internal/pkg/metrics/metrics.go

```diff
@@ -74,7 +74,7 @@ func (c *cacheHandler) Collect(ch chan<- prometheus.Metric) {
 			debugm("Debug:Cleaned up cache for host %s", key)
 		}
 	}
-
+	lock.Unlock()
```

Comment:
Instead of calling lock.Unlock() use defer
defer lock.Unlock() 
or 
defer func(){
    lock.Unlock() 
}